tests/unit/autojac/test_mtl_backward.py

```diff
@@ -541,3 +541,28 @@ def test_mtl_backward_task_params_are_subset_of_other_task_params():
 
     J = torch.tensor([[-p1, p1], [-p1 * p2, p1 * p2]], device=DEVICE)
     assert_close(p0.grad, A(J))
+
+
+def test_mtl_backward_shared_params_overlap_with_tasks_params():
+    """
+    Tests that mtl_backward raises an error when the set of shared params overlaps with the set of
+    task-specific params.
+    """
+
+    p0 = torch.tensor([1.0, 2.0], requires_grad=True, device=DEVICE)
+    p1 = torch.tensor(2.0, requires_grad=True, device=DEVICE)
+    p2 = torch.tensor(3.0, requires_grad=True, device=DEVICE)
+
+    r = torch.tensor([-1.0, 1.0], device=DEVICE) @ p0
+    y1 = r * p1
+    y2 = p0.sum() * r * p2
+
+    with raises(ValueError):
+        mtl_backward(
+            losses=[y1, y2],
+            features=[r],
+            A=UPGrad(),
+            tasks_params=[[p1], [p0, p2]],  # Problem: p0 is also shared
+            shared_params=[p0],
```

Comment:
No. The default value for `shared_params` should be `[p0]` and the default for `tasks_params` should be `[[p1], [p2]]`, because the `shared_params` should be explicitly excluded from each of the tasks' params.

We could add another check to verify what I just said.

Reply:
I just checked the code and we actually forgot to exclude `shared_params` from each task's params. I'll make another PR with this exclusion + the test I just mentioned.

Reply:
Isn't that supposed to be done in this line?
https://github.com/TorchJD/torchjd/blob/8ec5dcb3eaa84a49c72af7eae0121e491201be08/src/torchjd/autojac/mtl_backward.py#L89

Or maybe you are talking about excluding features from task's params when they are also leaves?